modules/features/manatime_feature_timesheets/includes/timesheet.inc

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/**
+ * @file
+ * Template file for timesheet.
+ */
+
+ /**
+  * Class timesheet.
+  */
+class Timesheet {
+
+  private $node;
+  private $user;
+  private $timeEntries;
+  private $projects = array();
+  private $status;
+  private $records;
+
+  /**
+   * Construct.
+   */
+  public function __construct($nid) {
+    $this->node = node_load($nid);
+    $this->user = $this->node->uid;
+    $this->status = $this->node->status;
+    $this->records = array();
+
+    $this->setTimeEntries();
+    $this->setProjects();
+  }
+
+  /**
+   * Get nid.
+   */
+  public function getNid() {
+    return $this->node->nid;
+  }
+
+  /**
+   * Set status.
+   */
+  public function setStatus($status) {
+    $this->status = $status;
+  }
+
+  /**
+   * Get status.
+   */
+  public function getStatus() {
+    return $this->status;
+  }
+
+  /**
+   * Get status.
+   */
+  public function getUser() {
+    return $this->user;
+  }
+
+  /**
+   * Get size.
+   */
+  public function getNumTimeEntries() {
+    return count($this->node->field_time_entries[LANGUAGE_NONE]);
+  }
+
+  /**
+   * Set time_entries.
+   */
+  public function setTimeEntries() {
+    if (isset($this->node->field_time_entries[LANGUAGE_NONE])) {
+      $result = $this->node->field_time_entries[LANGUAGE_NONE];
+      $nids = array();
+      foreach ($result as $record) {
+        $nids[] = $record['target_id'];
+      }
+      $this->timeEntries = node_load_multiple($nids);
+    }
+    else {
+      $this->timeEntries = array();
+    }
+  }
+
+  /**
+   * Get time_entries.
+   */
+  public function getTimeEntries() {
+    return $this->timeEntries;
+  }
+
+  /**
+   * Set projects.
+   */
+  public function setProjects() {
+    $nids = array();
+    foreach ($this->timeEntries as $entry) {
+      $nid = $entry->field_project[LANGUAGE_NONE][0]['target_id'];
+      if (!in_array($nid, $nids)) {
+        $nids[] = $nid;
+      }
+    }
+    $this->projects = node_load_multiple($nids);
+  }
+
+  /**
+   * Get projects.
+   */
+  public function getProjects() {
+    return $this->projects;
+  }
+
+  /**
+   * Get time_entries per day.
+   */
+  public function getTimeEntriesPerDay($day) {
+    $nids = array();
+    foreach ($this->timeEntries as $entry) {
+      if ($entry->field_date[LANGUAGE_NONE][0]['value'] === $day) {
+        $nids[] = $nid;
+      }
+    }
+    return node_load_multiple($nids);
```

Comment:
You're returning timeEntries, you shouldn't load them again, just pick the ones that correspond.